feat(ui/rate): allow to clear score

chouchouji opened the pull request:
## Checklist

---

- [ ] allow to clear score

nonzzz reviewed (approved):
LGTM. but short line is better isn't?

chouchouji commented:
> 

Sorry, I don't quite understand. Is it optimizing the code?

nonzzz commented:
@chouchouji 

Look like the follow block

```js
// better
if (condit  === xxx) doSometing 

// bad
 
if (condit === xxx) {
   doSometing 
}


```

If your logic is short. so is no need to wrap it in a block.

chouchouji commented:
> @chouchouji
> 
> Look like the follow block
> 
> ```js
> // better
> if (condit  === xxx) doSomething 
> 
> // bad
>  
> if (condit === xxx) {
>    doSomething 
> }
> ```
> 
> If your logic is short. so is no need to wrap it in a block.

I know it.  Yesterday when I was writing logic, I modified the logic in the if section, but in the end, I implemented it in another way. The original logic in the if section remained unchanged and I forgot to change it back. 
I will modify it later. Thank you!!!

chouchouji commented:
> @chouchouji
> 
> Look like the follow block
> 
> ```js
> // better
> if (condit  === xxx) doSomething 
> 
> // bad
>  
> if (condit === xxx) {
>    doSomething 
> }
> ```
> 
> If your logic is short. so is no need to wrap it in a block.

done

nonzzz commented:
@haoziqaq CC

chouchouji commented on packages/varlet-ui/src/rate/Rate.vue:
```diff
@@ -58,6 +58,7 @@ export default defineComponent({
     const { errorMessage, validateWithTrigger: vt, validate: v, resetValidation } = useValidation()
     const { hovering } = useHoverOverlay()
     const currentHoveringValue = ref<number>(-1)
+    const lastScore = ref<number>(Number(props.modelValue))
```
done
